sdk/eventhub/event-hubs/src/partitionReceiver.ts

```diff
@@ -351,28 +351,43 @@ export function waitForEvents(
     receivedAfterWait,
     receivedAlready,
   } = options;
+
+  if (queue.length >= maxEventCount) {
+    return Promise.resolve().then(receivedAlready);
+  }
+
   const aborter = new AbortController();
-  const { signal: abortSignal } = new AbortController([
-    aborter.signal,
-    ...(clientAbortSignal ? [clientAbortSignal] : []),
-  ]);
+  let abortListener: (() => void) | undefined;
+  if (clientAbortSignal) {
+    if (clientAbortSignal.aborted) {
+      throw new AbortError("The operation was aborted.");
+    }
+    abortListener = () => {
+      aborter.abort();
+    };
+    clientAbortSignal.addEventListener("abort", abortListener);
+  }
```

Comment:
I think this can be simplified to the following. Note that I removed the aborted check because it has already been done earlier:
```suggestion
  const abortListener = () => {
    aborter.abort();
  };
  clientAbortSignal?.addEventListener("abort", abortListener);
```